File: Leetcode/2427 Number of Common Factors.py
class Solution:
    def commonFactors(self, a, b):
        return divisor_count(gcd(a, b))


def gcd(a, b):
    return a if b == 0 else gcd(b, a % b)


def divisor_count(a):
    d, count = 1, 0
    while d * d <= a:
        if a % d == 0:
            count += 1
            other_d = a // d
            if d != other_d:
                count += 1
        d += 1
    return count

File: Leetcode/2427 Number of Common Factors.ts
function commonFactors(a: number, b: number): number {
    return nFactors(gcd(a, b));
}

function gcd(a: number, b: number): number {
    return b === 0 ? a : gcd(b, a % b);
}

function nFactors(n: number): number {
    let factors = 0;
    for (let d = 1; d * d <= n; d++) {
        if (n % d === 0) {
            factors++;
            const otherD = n / d;
            if (d !== otherD) {
                factors++;
            }
        }
    }
    return factors;
}
